server/uno.go

```diff
@@ -103,89 +95,60 @@ func contains(arr []string, val string) (int, bool) {
 ////////////////////////////////////////////////////////////
 // These are all of the functions for the game -> essentially public functions
 ////////////////////////////////////////////////////////////
-func updateGame(c echo.Context) *Response {
+func updateGame(game string, username string) bool {
 	success := false
-	if success = checkID(c.Param("game")); success && gameStarted {
-		return &Response{true, newPayload(c.Param("username"))}
+	if success = checkID(game); success && gameStarted {
+		return true
 	}
-	return &Response{false, nil}
+	return false
 }
 
-func createNewGame(c echo.Context) *Response {
-
-	ctx := context.Background()
-	client := createClient(ctx)
-
-	_, _, err := client.Collection("users").Add(ctx, map[string]interface{}{
-		"first": "Ada",
-		"last":  "Lovelace",
-		"born":  1815,
-	})
-	if err != nil {
-		log.Fatalf("Failed adding alovelace: %v", err)
-	}
-
+func createNewGame() string {
 	gameID = "12234"
-	return &Response{true, newPayload("")}
+	return gameID
 }
 
-func joinGame(c echo.Context) *Response {
-	ctx := context.Background()
-	client := createClient(ctx)
-
-	iter := client.Collection("users").Documents(ctx)
-	for {
-		doc, err := iter.Next()
-		if err == iterator.Done {
-			break
-		}
-		if err != nil {
-			log.Fatalf("Failed to iterate: %v", err)
-		}
-		fmt.Println(doc.Data())
-	}
-
-	if checkID(c.Param("game")) {
-		user := c.Param("username")
+func joinGame(game string, username string) bool {
+	if checkID(game) {
+		user := username
 		if _, found := contains(players, user); !found {
 			players = append(players, user)
 			allCards[user] = nil // No cards yet
 		}
-		return &Response{true, newPayload(c.Param("username"))}
+		return true
 	}
-	return &Response{false, nil} // bad game_id
+	return false // bad game_id
 }
 
-func playCard(c echo.Context, card Card) *Response {
-	if checkID(c.Param("game")) && currPlayer == c.Param("username") {
+func playCard(game string, username string, card Card) bool {
+	if checkID(game) && currPlayer == username {
 		if card.Color == currCard[0].Color || card.Number == currCard[0].Number {
 			// Valid card can be played
 			playerIndex = (playerIndex + 1) % len(players)
 			currPlayer = players[playerIndex]
 			currCard[0] = card
 
-			for index, item := range allCards[c.Param("username")] {
+			for index, item := range allCards[username] {
```

Comment:
Consider making `cards: := allCards[username]` as `allCards[username]` gets used  a lot here.

Reply:
I've gone ahead and made that change as well.